packages/builder/src/stores/builder/layouts.ts

```diff
@@ -69,6 +79,24 @@ export class LayoutStore extends BudiStore {
 
 export const layoutStore = new LayoutStore()
 
-export const selectedLayout = derived(layoutStore, $store => {
-  return $store.layouts?.find(layout => layout._id === $store.selectedLayoutId)
-})
+export class SelectedLayoutStore extends DerivedBudiStore<
+  LayoutState,
+  DerivedLayoutState
+> {
+  constructor(layoutStore: LayoutStore) {
+    const makeDerivedStore = () => {
+      return derived(layoutStore, $store => {
+        return {
+          ...$store,
+          selectedLayout:
+            $store.layouts?.find(
+              layout => layout._id === $store.selectedLayoutId
+            ) || null,
+        }
+      })
+    }
+    super(INITIAL_LAYOUT_STATE, makeDerivedStore)
+  }
+}
+
+export const selectedLayout = new SelectedLayoutStore(layoutStore)
```

Comment:
NAB: we can leave this as is since you've already converted it, but I think in this case we could leave the selected layout as a normal derived store rather than a derived budi store. It's just a simple piece of derived state rather than a full store with functions etc.

Reply:
I'm gonna move it back, it was much cleaner before!